state/src/impls/top_level.rs

```diff
@@ -999,6 +999,12 @@ impl TopState for TopLevelState {
         metadata.increase_seq();
         Ok(())
     }
+
+    fn snapshot_term_params(&mut self) -> StateResult<()> {
```

Comment:
I don't think snapshot is appropriate for this function. How about renaming it to `update_term_params`?

Reply:
Agreed. It was not my decision to name this function like this, but I think your suggestion is better.